admin/src/containers/Data/AttributesConfigContext.js

```diff
@@ -0,0 +1,67 @@
+import React, { Component, createContext } from 'react';
+import internalFetch from 'helpers/internal-fetch';
+import PropTypes from 'prop-types';
+import AppConfig from 'config';
+
+const AttributesConfigContext = createContext({
+  fetchConfig: () => {},
+  updateConfig: () => {},
+  config: null,
+  isBusy: false
+});
+
+export class AttributesConfigProvider extends Component {
+  static propTypes = {
+    children: PropTypes.oneOfType([
+      PropTypes.element,
+      PropTypes.object,
+      PropTypes.arrayOf(PropTypes.element)
+    ])
+  };
+
+  setBusy(busy) {
+    this.setState({
+      isBusy: busy
+    });
+  }
+
+  fetchConfig = async () => {
+    const { busy, config } = this.state;
+    if (busy || config) {
+      return;
+    }
+    this.setBusy(true);
+    const response = await internalFetch(
+      `${AppConfig.API_URL}/api/management/data/attributes-config`
+    );
+    this.setState({ config: response });
+    this.setBusy(false);
+  };
+
+  updateConfig = async config => {
+    this.setBusy(true);
+    await internalFetch(`${AppConfig.API_URL}/api/management/data/attributes-config/update`, {
+      method: 'POST',
+      body: JSON.stringify(config)
+    });
+    this.setState({ config });
+    this.setBusy(false);
```

Comment:
why not just change the state directly in the line above? this.setState({ isBusy: false }) is clear enough I think.

Reply:
That's true. I added this helper only to not type `isBusy` many times in different places, because it is possible to make a typo.